kinesisfirehose-logs-extension-demo/README.md

```diff
@@ -0,0 +1,226 @@
+Running NodejsNpmBuilder:NpmPack
+Running NodejsNpmBuilder:CopyNpmrc
+Running NodejsNpmBuilder:CopySource
+Running NodejsNpmBuilder:NpmInstall
+Running NodejsNpmBuilder:CleanUpNpmrc
+Building layer 'KinesisFireHoseLogsApiExtensionLayer'
+Running CustomMakeBuilder:CopySource
+Running CustomMakeBuilder:MakeBuild
+Current Artifacts Directory : /Users/xxx/CodeBase/aws-lambda-extensions/kinesisfirehose-logs-extension-demo/.aws-sam/build/KinesisFireHoseLogsApiExtensionLayer
+
+Build Succeeded
+
+Built Artifacts  : .aws-sam/build
+Built Template   : .aws-sam/build/template.yaml
+
+Commands you can use next
+=========================
+[*] Invoke Function: sam local invoke
+[*] Deploy: sam deploy --guided
+```
+
+### Deployment
+
+Run the following command to deploy the sample Lambda function with the extension
+
+```bash
+sam deploy --guided
+```
+
+The following parameters can be customized part of the deployment
+
+| Parameter  | Description | Default |
+| ------------- | ------------- | ----|
+| FirehoseStreamName  | Firehose stream name |  lambda-logs-direct-s3-no-cloudwatch |
+| FirehoseS3Prefix  | The S3 Key prefix for Kinesis Firehose |  lambda-logs-direct-s3-no-cloudwatch |
+| FirehoseCompressionFormat  | Compression format used by Kinesis Firehose, allowed value - `UNCOMPRESSED, GZIP, Snappy` |  GZIP |
+| FirehoseBufferingInterval  | How long Firehose will wait before writing a new batch into S3 |  60 |
+| FirehoseBufferingSize  | Maximum batch size in MB |  10 |
+
+> Note: We can either customize the parameters, or leave it as default to proceed with the deployment
+
+**Output**
+
+```bash
+CloudFormation outputs from deployed stack
+-------------------------------------------------------------------------------------------------------------------
+Outputs
+-------------------------------------------------------------------------------------------------------------------
+Key                 KinesisFireHoseLogsApiExtensionLayer
+Description         Kinesis Log emiter Lambda Extension Layer Version ARN
+Value               arn:aws:lambda:us-east-1:xxx:layer:kinesisfirehose-logs-extension-demo:5
+
+Key                 BucketName
+Description         The bucket where data will be stored
+Value               sam-app-deliverybucket-xxxx
+
+Key                 KinesisFireHoseIamRole
+Description         Kinesis firehose IAM role
+Value               arn:aws:firehose:us-east-1:xxx:deliverystream/lambda-logs-direct-s3-no-cloudwatch
+
+Key                 HelloWorldFunction
+Description         First Lambda Function ARN
+Value               arn:aws:lambda:us-east-1:xxx:function:kinesisfirehose-logs-extension-demo-function
+-------------------------------------------------------------------------------------------------------------------
+```
+
+## Testing
+
+You can invoke the Lambda function using the following CLI command
```

Comment:
Suggestion: "You can invoke the Lambda function using the [Lambda Console](https://console.aws.amazon.com/lambda/home), or the following CLI command"